src/Components/WebAssembly/WebAssembly/src/Services/WebAssemblyLoggerFactory.cs

```diff
@@ -1,19 +1,29 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging;
+using Microsoft.JSInterop;
 
 namespace Microsoft.AspNetCore.Components.WebAssembly.Services
 {
     internal class WebAssemblyLoggerFactory : ILoggerFactory
     {
+        private readonly IJSInProcessRuntime _jsRuntime;
+
+        public WebAssemblyLoggerFactory(IServiceProvider services)
+        {
+            _jsRuntime = (IJSInProcessRuntime)services.GetRequiredService<IJSRuntime>();
+        }
+
         public void AddProvider(ILoggerProvider provider)
         {
             // No-op
```

Comment:
I understand that this is not part of this PR, but shouldn't we implement this factory properly?

I believe the existing pattern has extension methods on the factory to add different providers. I'm fine if we file a follow-up issue for this, but I don't want it to get lost in the ether

Reply:
This probably also needs to respect login levels comming from config (as asp.net core does)

Reply:
Filed https://github.com/dotnet/aspnetcore/issues/19737